File: Hackethon/ArrayLeftRotation.py
""" Sample Input
5 4
1 2 3 4 5
Sample Output
5 1 2 3 4  """

def rotLeft(arr, d):
    # Write your code here
    n = len(arr)
    print(d % n)

    rotated_arr = arr[d :] + arr[:d ]
    return rotated_arr

# Example usage:
original_array = [1, 2, 3, 4, 5]
d_steps = 4

rotated_array = rotLeft(original_array, d_steps)

print("Original Array:", original_array)
print(f"Left rotated by {d_steps} steps:", rotated_array)

File: Hackethon/ArrayLeftRotation.java
package Hackethon;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

/* Sample Input
5 4
1 2 3 4 5
Sample Output
5 1 2 3 4 */
public class ArrayLeftRotation {
        public static List<Integer> rotLeft(List<Integer> a, int d) {
            List<Integer> rotation = new ArrayList<>();
            for(int i = d; i<a.size(); i++){
                rotation.add(a.get(i));
            }
            for(int i = 0; i< d; i++){
                rotation.add(a.get(i));
            }
            return rotation;
        }

        public static void main(String[] args) {
            List<Integer> arr = new ArrayList<>(Arrays.asList(1, 2, 3, 4, 5));
            System.out.println(rotLeft(arr, 4));
        }
}
